boilerplate/_lib/boilerplate-commit

```diff
@@ -0,0 +1,185 @@
+  # Generate the commit history for this range. This will go in the commit message.
+  (
+    if [ -z "$BOILERPLATE_GIT_REPO" ]; then
+      BOILERPLATE_GIT_REPO=git@github.com:openshift/boilerplate.git
+    fi
+    if [ -z "$BOILERPLATE_GIT_CLONE" ]; then
+      BOILERPLATE_GIT_CLONE="git clone $BOILERPLATE_GIT_REPO"
+    fi
+    $BOILERPLATE_GIT_CLONE $bp_clone
+    cd $bp_clone
+    # Matches promote.sh
+    git log --no-merges --pretty=format:'commit: %H%nauthor: %an%n%s%n%n%b%n%n' $bp_commit_change > $bp_log
+  )
+
+fi
+
+# Okay, let's look for convention changes.
+# TODO: Handle unsubscribes (not yet handled by the main `update` either).
+while read convention junk; do
+  # TODO: These first few conditions, scrubbing the config file, are
+  # identical to what's in `update`. It would be lovely to library-ize
+  # them. However, `update` needs to remain self-contained since it's
+  # part of the bootstrap process.
+
+  # Skip comment lines (which can have leading whitespace)
+  if [[ "$convention" == '#'* ]]; then
+    continue
+  fi
+  # Skip blank or whitespace-only lines
+  if [[ "$convention" == "" ]]; then
+    continue
+  fi
+  # Lines like
+  #    valid/path  other_junk
+  # are not acceptable, unless `other_junk` is a comment
+  if [[ "$junk" != "" ]] && [[ "$junk" != '#'* ]]; then
+    echo "Invalid config! Only one convention is allowed per line. Found '$junk'. Ignoring."
+    # `update` bails for this. We're being a bit more forgiving.
+    continue
+  fi
+
+  dir_path="boilerplate/${convention}"
+  # Make sure the directory exists
+  if ! [[ -d "$dir_path" ]]; then
+    echo "Invalid convention directory: '$convention'."
+    echo "(Could be because we don't handle unsubscribing yet.)"
+    echo "Ignoring."
+    # `update` bails for this. We're being a bit more forgiving.
+    continue
+  fi
+
+  # Okay, we have a legit convention. Let's see if the current checkout
+  # touches it
+  # (Note that we're reusing the same temp file on each iteration.)
+  git status --porcelain $dir_path > $convention_status
+  if ! [[ -s $convention_status ]]; then
+    # No deltas here.
+    echo "- $convention: No change" >> $convention_statuses
+
+  elif grep -q -v '^??' $convention_status; then
+    # If there's anything *other than* untracked, this was an update
+    echo "- $convention: Update" >> $convention_statuses
+
+  else
+    # If we get here, everything is '^??' (untracked), meaning this is a
+    # new subscription. (Or, I suppose, the convention was previously
+    # empty? We'll call it a new subscription anyway.)
+    echo "- $convention: Subscribe" >> $convention_statuses
+  fi
+
+done < boilerplate/update.cfg
+
+# Let's make sure *something* boilerplate-related is happening here.
+if [[ -z "$bootstrap" ]] && [[ -z "$bp_commit_change" ]] && ! grep -v -q "No change" $convention_statuses; then
+  err "No boilerplate-related activity found in the current checkout!"
+fi
+
+# Okay, we're ready to do this.
```

Comment:
Would be good to explain what the Title/branch name looks like

Reply:
Will do.

Reply:
Done.